homeassistant/components/zerproc/light.py

```diff
@@ -116,11 +116,14 @@ async def async_added_to_hass(self) -> None:
 
     async def async_will_remove_from_hass(self) -> None:
         """Run when entity will be removed from hass."""
-        await self._light.disconnect()
+        try:
+            await self._light.disconnect()
+        except pyzerproc.ZerprocException:
+            pass
 
     async def on_hass_shutdown(self, event):
         """Execute when Home Assistant is shutting down."""
-        await self._light.disconnect()
+        await self.async_will_remove_from_hass()
```

Comment:
Why not call `async_will_remove_from_hass` directly? This is a single-line proxy method now...

Reply:
Good catch! Missed that in the refactoring process.